src/github.com/t3chguy/riot-static/riot-static.go

```diff
@@ -1,232 +0,0 @@
-// Copyright 2017 Michael Telatynski <7t3chguy@gmail.com>
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package main
-
-import (
-	"fmt"
-	"github.com/gin-gonic/gin"
-	"github.com/t3chguy/riot-static/matrix-client"
-	"github.com/t3chguy/riot-static/utils"
-	"net/http"
-	"os"
-	"strconv"
-	"time"
-)
-
-// TODO Cache memberList+serverList until it changes
-
-const PublicRoomsPageSize = 20
-const RoomTimelineSize = 20
-const RoomMembersPageSize = 20
-
-func LoadPublicRooms(first bool) {
-	fmt.Println("Loading publicRooms")
-	resp, err := client.PublicRooms(0, "", "")
-
-	if err != nil {
-		// Only panic if first one fails, after that we only have outdated data (less important)
-		if first {
-			panic(err)
-		} else {
-			fmt.Println(err)
-		}
-	}
-
-	// Preallocate the maximum capacity possibly needed (if all rooms were world readable)
-	worldReadableRooms := make([]*matrix_client.Room, 0, len(resp.Chunk))
```

Comment:
I don't really know why you're bothering with this. Most of the time this will be way oversized (since most rooms will not be world readable). In fact, provided <50% of rooms are world readable, this will always be allocating more than the naive `append()` sizing (doubling if reached cap).

It doesn't really matter anyway, given we're dealing with a slice of pointers...